src/Diagnostics.DataProviders/KustoHeartBeatService/KustoHeartBeatService.cs

```diff
@@ -52,17 +53,48 @@ public KustoHeartBeatService(KustoDataProviderConfiguration configuration)
             Initialize();
         }
 
-        public string GetClusterNameFromStamp(string stampName)
+        public async Task<string> GetClusterNameFromStamp(string stampName)
         {
             string kustoClusterName = null;
             string appserviceRegion = ParseRegionFromStamp(stampName);
 
             if (!_configuration.RegionSpecificClusterNameCollection.TryGetValue(appserviceRegion.ToLower(), out kustoClusterName))
             {
+                if(_configuration.CloudDomain == DataProviderConstants.AzureCloud)
+                {
+                    try
+                    {
+                         DataTable regionMappings = await _kustoDataProvider.ExecuteClusterQuery($"cluster('wawseusfollower').database('wawsprod').WawsAn_regionsincluster | where pdate >= ago(5d) | summarize by Region, ClusterName", "RegionMappingInit");
+                        if (regionMappings.Rows.Count > 0)
+                        {
+                            foreach (DataRow dr in regionMappings.Rows)
+                            {
+                                _configuration.RegionSpecificClusterNameCollection.TryAdd(((string)dr["Region"]).ToLower(), ((string)dr["ClusterName"]).ToLower());
+                                _configuration.FailoverClusterNameCollection.TryAdd(((string)dr["Region"]).ToLower(), $"{((string)dr["ClusterName"])}follower");
```

Comment:
Switch this. Follower clusters is diagnostics main clusters and vice versa for the failover clusters.